cmd/bcpd/app/app_test.go

```diff
@@ -373,3 +436,103 @@ func queryAndCheckToken(t *testing.T, baseApp app.BaseApp, path string, data []b
 		require.Equal(t, expected[i], actual)
 	}
 }
+
+// makeSendTx is a special case of sendToken when the sender account is the only signer
+// this is used in our benchmark
+func makeSendTx(t require.TestingT, chainID string, sender, receiver *account, ticker, memo string, amount int64) []byte {
+	msg := &cash.SendMsg{
+		Src:  sender.address(),
+		Dest: receiver.address(),
+		Amount: &x.Coin{
+			Whole:  amount,
+			Ticker: ticker,
+		},
+		Memo: memo,
+	}
+
+	tx := &Tx{
+		Sum: &Tx_SendMsg{msg},
+	}
+
+	sig, err := sigs.SignTx(sender.pk, tx, chainID, sender.nonce())
+	require.NoError(t, err)
+	tx.Signatures = append(tx.Signatures, sig)
+	txBytes, err := tx.Marshal()
+	require.NoError(t, err)
+	require.NotEmpty(t, txBytes)
+	return txBytes
+}
+
+// benchmarkSendTx runs the actual benchmark sequence eg.
+// N * CheckTx
+// BeginBlock
+// N * DeliverTx
+// EndBlock
+// Commit
+func benchmarkSendTx(b *testing.B, nbAccounts, blockSize int) {
+	accounts := make([]*account, nbAccounts)
+	for i := 0; i < nbAccounts; i++ {
+		accounts[i] = &account{pk: crypto.GenPrivKeyEd25519()}
+	}
+
+	chainID := "bench-net-22"
+	myApp := newTestApp(b, chainID, accounts)
+
+	txs := make([][]byte, b.N)
+	for i := 1; i <= b.N; i++ {
+		sender := accounts[cmn.RandInt()%nbAccounts]
+		recipient := accounts[cmn.RandInt()%nbAccounts]
+		txs[i-1] = makeSendTx(b, chainID, sender, recipient, "ETH", "benchmark", 1)
+	}
+
+	b.ResetTimer()
+
+	for i := 1; i <= b.N; i++ {
+		chres := myApp.CheckTx(txs[i-1])
+		require.Equal(b, uint32(0), chres.Code, chres.Log)
+	}
+
+	k := 1
+	header := abci.Header{Height: int64(k)}
+	myApp.BeginBlock(abci.RequestBeginBlock{Header: header})
+	for i := 1; i <= b.N; i++ {
+		if i%blockSize == 0 {
+			myApp.EndBlock(abci.RequestEndBlock{})
+			cres := myApp.Commit()
+			assert.NotEmpty(b, cres.Data)
+
+			k++
+			header = abci.Header{Height: int64(k)}
+			myApp.BeginBlock(abci.RequestBeginBlock{Header: header})
+		}
+
+		dres := myApp.DeliverTx(txs[i-1])
+		require.Equal(b, uint32(0), dres.Code, dres.Log)
+	}
+
+	myApp.EndBlock(abci.RequestEndBlock{})
+	cres := myApp.Commit()
+	assert.NotEmpty(b, cres.Data)
+}
+
+// Runs benchmarks with various input combination of initial accounts and block size
+func BenchmarkSendTx(b *testing.B) {
+	benchmarks := []struct {
+		accounts  int
+		blockSize int
+	}{
+		{100, 10},
+		{100, 200},
+		{10000, 1000},
+		{10000, 2000},
+		{100000, 1000},
+		{100000, 10000},
```

Comment:
Honestly, I think blockSize of more than 1000 tx is unlikely....
Maybe sizes of 10, 100, 1000 for every account size, to better compare them